src/main/java/core/basesyntax/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,57 @@
+package core.basesyntax.dao.impl;
+
+import core.basesyntax.dao.ManufacturerDao;
+import core.basesyntax.db.Storage;
+import core.basesyntax.lib.Dao;
+import core.basesyntax.model.Manufacturer;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (Manufacturer manufacturer : manufacturers) {
+            if (manufacturer.getId() == id) {
+                return Optional.of(manufacturer);
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        return Storage.manufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        List<Manufacturer> manufacturers = Storage.manufacturers;
+        for (Manufacturer object : manufacturers) {
+            if (object.getId() == manufacturer.getId()) {
+                Storage.manufacturers.set(Math.toIntExact(object.getId() - 1L), manufacturer);
```

Comment:
it's up to you, but method set() have int as parameterm and when you have more digital than int you will get some erros. I would write like this: 
```suggestion
                Storage.manufacturers.remove(object.getId());
                Storage.manufacturers.add(manifacturer);